src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,616 @@
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_GHOST_VISUAL:
+                    DoCast(me, SPELL_GHOST_DISGUISE);
+                    _events.ScheduleEvent(EVENT_REMOVE_GHOST_VISUAL, 6000, 0, PHASE_TWO);
+                    break;
+                case EVENT_REMOVE_GHOST_VISUAL:
+                    me->RemoveAura(SPELL_GHOST_DISGUISE);
+                    _events.ScheduleEvent(EVENT_GHOST_VISUAL, 9000, 0, PHASE_TWO);
+                    break;
+                case EVENT_RESURFACE_SOON:
+                    Talk(EMOTE_EMERGE_SOON);
+                    _events.CancelEvent(EVENT_GHOST_VISUAL);
+                    _events.CancelEvent(EVENT_REMOVE_GHOST_VISUAL);
+                    break;
+                }
+            }
+        }
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_frozen_coreAI(creature);
+    }
+};
+
+class npc_ahune_bunny : public CreatureScript
+{
+public:
+    npc_ahune_bunny() : CreatureScript("npc_ahune_bunny") { }
+
+    struct npc_ahune_bunnyAI : public ScriptedAI
+    {
+        npc_ahune_bunnyAI(Creature* creature) : ScriptedAI(creature), _summons(me)
+        {
+            instance = me->GetInstanceScript();
+            count = 0;
+            say_timer = 3000;
+        }
+
+        InstanceScript* instance;
+        uint32 count;
+        uint64 say_timer;
+        SummonList _summons;
+
+        void JustSummoned(Creature* summon) override
+        {
+            _summons.Summon(summon);
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _summons.DespawnAll();
+        }
+
+        void DoAction(int32 action)
+        {
+            if (action = ACTION_START_EVENT)
+            {
+                TempSummon* sliperryFloor = me->SummonCreature(NPC_SLIPPERY_FLOOR, SummonPositions[7], TEMPSUMMON_MANUAL_DESPAWN);
+                sliperryFloor->SetReactState(REACT_PASSIVE);
+                sliperryFloor->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+                sliperryFloor->AddAura(SPELL_SLIPPERY_FLOOR_AMBIENT, sliperryFloor);
+                for (int counter = 1; counter < 7; counter++)
+                {
+                    if (counter < 4)
+                        me->SummonCreature(NPC_EARTHEN_RING_TOTEM, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                    else
+                        me->SummonCreature(NPC_EARTHEN_RING_FLAMECALLER, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                }
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (say_timer <= diff)
```

Comment:
This whole thing should propably be handled as periodic ticks of http://www.wowhead.com/spell=45928 also it's not supposed to be a random player but the one who activates the gossip (in go_ahune_ice_stone) and the timing is (atleast on 6.1.2) 1 sec between those say lines.